app/jobs/course/user_deletion_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+class Course::UserDeletionJob < ApplicationJob
+  include TrackableJob
+  queue_as :highest
+
+  protected
+
+  def perform_tracked(course, course_user, current_user)
+    @course = course
+    @course_user = course_user
+    @current_user = current_user
+
+    ActsAsTenant.without_tenant do
+      success = course_user.class.transaction do
+        raise ActiveRecord::Rollback unless course_user.destroy
+
+        true
+      end
+
+      notify_deletion_failure unless success
+    end
+  end
+
+  def notify_deletion_failure
+    Course::Mailer.
+      course_user_deletion_failed_email(@course, @course_user, @current_user).
+      deliver_now
+  end
```

Comment:
Add a test to `Course::UsersController` to check for failure email

Reply:
I'm adding the said test to Course::UserDeletionJobSpec instead